### PR TITLE
[TW-84247] Windows-based Docker images: migrate from Windows 2004 to Windows 2022

### PR DESCRIPTION
**Pull request by AndreyKoltsov1997**
As per Microsoft's release statement [Windows 10, version 2004 end of servicing (all editions)](https://learn.microsoft.com/en-us/lifecycle/announcements/windows-10-version-2004-end-of-servicing), version 2004 is no longer supported:

Windows 10, version 2004 will reach the end of servicing on December 14, 2021.

As a result, we must eleminate its usage from TeamCity Docker Images and replace it with [Windows Server 2022](https://learn.microsoft.com/en-us/windows/release-health/windows-server-release-info), whose mainstream end of support is scheduled to October of 2026.

Pull request includes:
* Implementation of Dockerfiles, along with the mandatory changes within Dockerfiles generator;
* Automation of the build process;
* Documentation updates;

### CODE REVIEW

File: configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+# Based on ${powershellImage} 3
+FROM ${powershellImage} AS dotnet
+
+# On some agents, Windows 2022 requires administrator permissions to modify "C:/" folder within ...
+# ... PowerShell container.
+USER ContainerAdministrator
+
+COPY scripts/*.cs /scripts/
+SHELL ["pwsssssssh", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
```
Comment (burnasheva) [inline configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile:29]: pwsh?

File: configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+COPY --from=tools /BuildAgent /BuildAgent
+
+EXPOSE 9090
+
+VOLUME C:/BuildAgent/conf
+
+    # Configuration file for TeamCity agent
+ENV CONFIG_FILE="C:/BuildAgent/conf/buildAgent.properties" \
+    # Java home directory
+    JAVA_HOME="C:\Program Files\Java\OpenJDK" \
```
Comment (burnasheva) [inline configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile:80]: Why not C:/Program Files/Java/OpenJDK as we have two lines above?

File: configs/windows/Agent/windowsservercore/WindowsServerCore2022.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+EXPOSE 9090
+
+VOLUME C:/BuildAgent/conf
+
+CMD ["pwsh", "./BuildAgent/run-agent.ps1"]
+
+    # Configuration file for TeamCity agent
+ENV CONFIG_FILE="C:/BuildAgent/conf/buildAgent.properties" \
+    # Java home directory
+    JAVA_HOME="C:\Program Files\Java\OpenJDK" \
```
Comment (burnasheva) [inline configs/windows/Agent/windowsservercore/WindowsServerCore2022.Dockerfile:91]: Same here, it seems that it's better to use \ instead of /

**Review by burnasheva — CHANGES_REQUESTED** [review]

File: configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile (hunk 1 above)
Comment (AndreyKoltsov1997) [reply configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile]: This is a great catch, thank you! 
Please, excuse me - the line was accidentally added when from the upstream pull request, specifically: the implementation of .NET-based generation of Dockerfiles for the new images.
Fixed it.

File: configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile (hunk 2 above)
Comment (AndreyKoltsov1997) [reply configs/windows/Agent/nanoserver/NanoServer2022.Dockerfile:80]: The PowerShell environment within our base images is slash-agnostic, accepting both forward and backward slashes.

Given that the majority of the paths have backward slashes, I've aligned the `CONFIG_FILE` path with this format for consistency. The change had been applied to all Dockerfiles for Windows-based images within the repository.

```
…
ENV CONFIG_FILE="C:\BuildAgent\conf\buildAgent.properties" \
…
```

**Review by burnasheva — APPROVED** [review]

**Pull request merged**